pkg/asset/manifests/topologies.go

```diff
@@ -30,7 +30,12 @@ func determineTopologies(installConfig *types.InstallConfig) (controlPlaneTopolo
 
 	switch numOfWorkers {
 	case 0:
-		infrastructureTopology = controlPlaneTopology
+		// Two node deployments regardless of worker count require HA for InfraTopology
+		if controlPlaneReplicas == 2 {
+			infrastructureTopology = configv1.HighlyAvailableTopologyMode
+		} else {
+			infrastructureTopology = controlPlaneTopology
+		}
 	case 1:
 		infrastructureTopology = configv1.SingleReplicaTopologyMode
```

Comment:
> // Two node deployments regardless of worker count require HA for InfraTopology

@eggfoobar If `numOfWorkers == 1`, infraTopology will be set to `SingleReplicaTopologyMode`. That would violate the requirement in this comment? Or am I misunderstanding something 😅?

Reply:
@tthvo numOfWorkers == 1 is an undefined state of being for the two node variants, which currently do not support compute nodes.

IMHO, if you have a compute node, it's fair to expect the infra services to be running on them, so this logic is sensible.

With 0 compute nodes, your only option is to treat your CP as workers, and thus we need to differentiate based on compute node count.

Reply:
Oh okay, that makes sense. I was a bit unsure about the comment. Thanks @jaypoulz!

Reply:
Yeah, funny that you pointed that out, I was just thinking that comment is too strict about what we want here, updating it now as well.

Reply:
Sounds good! Thanks!